testsuite/MDAnalysisTests/analysis/test_align.py

```diff
@@ -151,6 +151,19 @@ def test_toggle_atom_nomatch_mismatch_atoms(self, universe, reference):
         with pytest.raises(SelectionError):
             align.alignto(u, ref, select='all', match_atoms=False)
 
+    def test_subselection_alignto(self, universe, reference):
+        sub_string = 'resname ALA and name CA'
+        rmsd = align.alignto(universe, reference, subselection=sub_string)
+        assert_almost_equal(rmsd[1], 5.27e-7, decimal=9)
```

Comment:
I see this one is failing in CI. A bit strange at first glance since it passes for me locally, whether I run the test class or just reproduce the code outside of the testing framework.

Thinking a little more, since the two universes are identical, I think this is basically applying the identity rotation matrix to the `subselection`, so maybe there's an instability there in some cases?

Even when the tests passes locally, the `old_rmsd` is 0.0 while the `new_rmsd` is the `5.27..e-7`. I can't decide if that makes sense or not--nothing should change unless the comparison is only using the subselection after, but then what is the subselection compared with?

Reply:
There's a line of Cython code in `MDAnalysis/lib/qcprot.pyx`

`rms = sqrt(fabs(2.0 * (E0 - mxEigenV)/N))`

Where the square root of `2.78755634425465e-13` is taken to produce the `5.27.. e-7` number. That's inside `FastCalcRMSDAndRotation()` function. Maybe there's a precision guard in there that needs a bit of thought.

Reply:
@tylerjereddy I recently implemented the QCP method and noticed this problem as well. For identical structures we should have `E0 = mxEigenV`, therefore I think that because of small numerical errors in `mxEigenV`,  they end up being slightly different. I was actually getting a negative number there, which is how I spotted the problem.

Reply:
Here's a diff I used locally to fix the issue and get this PR to pass tests on the expectation there is `0.0` rmsd here:

```diff
diff --git a/package/MDAnalysis/analysis/align.py b/package/MDAnalysis/analysis/align.py
index 04101b7d7..b694abd48 100644
--- a/package/MDAnalysis/analysis/align.py
+++ b/package/MDAnalysis/analysis/align.py
@@ -271,6 +271,9 @@ def rotation_matrix(a, b, weights=None):
     if a.shape != b.shape:
         raise ValueError("'a' and 'b' must have same shape")

+    if np.allclose(a, b) and weights is None:
+        return np.eye(3, dtype=np.float64), 0.0
+
     N = b.shape[0]

     if weights is not None:
diff --git a/testsuite/MDAnalysisTests/analysis/test_align.py b/testsuite/MDAnalysisTests/analysis/test_align.py
index a0a202863..0fbe8dc0a 100644
--- a/testsuite/MDAnalysisTests/analysis/test_align.py
+++ b/testsuite/MDAnalysisTests/analysis/test_align.py
@@ -154,11 +154,11 @@ class TestGetMatchingAtoms(object):
     def test_subselection_alignto(self, universe, reference):
         sub_string = 'resname ALA and name CA'
         rmsd = align.alignto(universe, reference, subselection=sub_string)
-        assert_almost_equal(rmsd[1], 5.27e-7, decimal=9)
+        assert_almost_equal(rmsd[1], 0.0, decimal=9)

         sub_ag = universe.select_atoms('resname ALA and name CA')
         rmsd = align.alignto(universe, reference, subselection=sub_ag)
-        assert_almost_equal(rmsd[1], 5.27e-7, decimal=9)
+        assert_almost_equal(rmsd[1], 0.0, decimal=9)

         with pytest.raises(TypeError):
             random_no = 9821
```

If other core devs agree with that (or similar) it may dislodge this PR. I also note that MDA seems to be a bit particular about identical structures when weights are specified (that doesn't always give `0.0` rmsd); for now I've said just "no weights and identical structures means 0.0 rmsd."

Reply:
@tylerjereddy so how should I proceed from here?

Reply:
like I got your point that since both `universe` and `reference` is identical then using `subselection` their `new_rmsd` value must be `0.0` but it is not. Should I change the mentioned changes in `rotation_matrix` function in my PR?

Reply:
I'd suggest making those changes yes.